x/gov/interfaces.go

```diff
@@ -0,0 +1,40 @@
+package gov
+
+import (
+	weave "github.com/iov-one/weave"
+)
+
+// OptionDecoder is needed to parse the raw_options data
+type OptionDecoder func(raw []byte) (weave.Msg, error)
```

Comment:
The is only a decoder. How are the options encoded? I think those two be provided together. Maybe an interface with `Decode` and `Encode` methods (or `Marshal` and `Unmarshal` as many people confuse the first pair) should be declared and used instead?

Reply:
Encoded options are passed into the messages. That is done client side. The server only needs to Decode.

https://github.com/iov-one/weave/blob/d7f6bab1b37602118f89fde79516c4a6d21d6d10/x/gov/sample_code_test.go#L9-L18 is a sample of how to decode

https://github.com/iov-one/weave/blob/d7f6bab1b37602118f89fde79516c4a6d21d6d10/x/gov/sample_test.proto#L26-L31 is a sample of how a client would form the msg (not needing decode then)